document.open() should remove the initial about:blank-ness of the document

cdumez opened the pull request:
#### dc6d5ef7e6246b46f5ea591be485df48735dfd01
<pre>
document.open() should remove the initial about:blank-ness of the document
<a href="https://bugs.webkit.org/show_bug.cgi?id=245445">https://bugs.webkit.org/show_bug.cgi?id=245445</a>

Reviewed by Darin Adler and Youenn Fablet.

document.open() should remove the initial about:blank-ness of the document.
This aligns our behavior with Firefox and Chrome.

* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/remove-initial-about-blankness.window-expected.txt:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::open):

Canonical link: <a href="https://commits.webkit.org/254747@main">https://commits.webkit.org/254747@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b123149d049cdb3b872989f7331194c2940c7321

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90072 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34619 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20696 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/99396 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/156901 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/94078 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33110 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/82402 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93692 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95720 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26319 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76877 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26230 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81175 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80964 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69231 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/34294 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15045 "Passed tests") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/32135 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15987 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3335 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/35879 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38956 "Passed tests") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/65/builds/37780 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35071 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3043,8 +3043,16 @@ ExceptionOr<void> Document::open(Document* entryDocument)
-    if (m_frame)
+    if (m_frame) {
+        // Calling window.open() counts as committing the first real document load.
+        WTFLogAlways("CHRIS: state was %u", (unsigned)m_frame->loader().stateMachine().stateForDebugging());
```
Probably do not want those logs

youennf commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3043,8 +3043,16 @@ ExceptionOr<void> Document::open(Document* entryDocument)
 
-    if (m_frame)
+    if (m_frame) {
+        // Calling window.open() counts as committing the first real document load.
```
Can we add a pointer to the spec, either here or in commit message

darinadler commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3043,8 +3043,16 @@ ExceptionOr<void> Document::open(Document* entryDocument)
+    if (m_frame) {
+        // Set document's is initial about:blank to false.
+        // https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#document-open-steps (step 13)
+        if (m_frame->loader().stateMachine().committingFirstRealLoad())
```
This looks like maybe it should be a `FrameLoader` function.

cdumez committed:
document.open() should remove the initial about:blank-ness of the document
https://bugs.webkit.org/show_bug.cgi?id=245445

Reviewed by Darin Adler and Youenn Fablet.

document.open() should remove the initial about:blank-ness of the document.
This aligns our behavior with Firefox and Chrome.

* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/remove-initial-about-blankness.window-expected.txt:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::open):

Canonical link: https://commits.webkit.org/254747@main
